selfdrive/updated.py

```diff
@@ -256,8 +258,19 @@ def finalize_update() -> None:
   run(["git", "reset", "--hard"], FINALIZED)
   run(["git", "submodule", "foreach", "--recursive", "git", "reset"], FINALIZED)
 
-  set_consistent_flag(True)
-  cloudlog.info("done finalizing overlay")
+  cloudlog.info("Starting git gc")
+  t = time.monotonic()
+  try:
+    run(["git", "gc"], FINALIZED)
+    cloudlog.info(f"Done git gc, took {time.monotonic() - t:.3f} s")
+  except subprocess.CalledProcessError:
```

Comment:
@adeebshihadeh Do we want to block updating if `git gc` fails or is that too risky? Currently this `except` triggers when going onroad during the `git gc`, which is why I have to check the wait helper afterwards to not finalize the overlay in a half-finished gc state.

Reply:
hmm, can interruption corrupt the repo? you almost want an overlay for the overlay :P

Reply:
Can't find anything definitive about corruption. Think it's not a big deal to start over if you interrupt it.